src/R/Components/Impl/ConnectionManager/Implementation/Connection.cs

```diff
@@ -20,5 +30,43 @@ internal class Connection : ConnectionInfo, IConnection {
         /// If true, the connection is to a remote machine
         /// </summary>
         public bool IsRemote { get; }
+
+        public override string UserProvidedPath {
+            get { return base.UserProvidedPath; }
+            set {
+                base.UserProvidedPath = value;
+                Path = ToCompletePath(value);
+            }
+        }
+
+        public static string ToCompletePath(string path) {
```

Comment:
I think this should be part of `ConnectionViewModel`, and change should be visible to the user (that is at least what we have discussed online). I'd expect to change the URL when path input field loses focus.